app/controllers/event_bookings_controller.rb

```diff
@@ -0,0 +1,55 @@
+class EventBookingsController < ApplicationController
+  # To-Do : Use Pundit for better authorization
+	before_action :authenticate_user!, except: [:index, :home,  :show]
+  before_action :set_event_booking, only: [:show, :update, :destroy]
+
+  def index
+    @event_bookings = EventBooking.includes(:user, :event).all
+    #Implement searching and pagination using Pagy gem
+  end
+
+  def show
+  end
+
+  def create
+    @event_booking = EventBooking.new(event_booking_params)
+    @event_booking.user_id = current_user.id
+    @event_booking.event_id = params[:event_id]
+    respond_to do |format|
+      if @event_booking.save
+        render notice: t("resource.created", { resource_name: "Event booking" })
+      else
+        format.json { render json: @event_booking.errors, status: :unprocessable_entity }
+      end
+    end
+  end
+
+  def update
+    respond_to do |format|
+      if @event_booking.update(event_booking_params)
```

Comment:
We can handle the case of booking cancellation and release the ticket, when some user cancels their booking given their cancellation comes before booking.

i.e. Extracting logic to service
```ruby

module EventBookings
  class UpdateService 
      attr_reader :event_booking, :event, :user, :response

    def initialize(event_booking)
      @event_booking = event_booking
      @event = event_booking.event
    end

    def process!
      if @event_booking.update(event_booking_params)
         notify_visitor
         response = { success: true }
      else
        response = { success: false }
      end
    end

    private

      def notify_visitor
        NotificationService.new(event_booking:).process
      end
  end
end
```

NotificationService handles mail, mobile notifications, or integration effects actions on EventBooking changes like cancellation of ticket, booking date shifts, venue switch requests etc.